packages/react-core/src/components/FileUpload/examples/FileUploadCustomPreview.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { FileUpload } from '@patternfly/react-core';
+import FileUploadIcon from '@patternfly/react-icons/dist/esm/icons/file-upload-icon';
+
+export const CustomPreviewFileUpload: React.FunctionComponent = () => {
+  const [value, setValue] = React.useState(null);
+  const [filename, setFilename] = React.useState('');
+
+  const handleFileInputChange = (_event, file) => {
```

Comment:
Could you also add types to the parameters of all these helper functions that are passed as props to the component? It helps us check that our components are handling the prop types correctly and helps communicate more clearly to the consumers which types they should be using.

Reply:
Sounds great! I just typed all of the helper functions :D